src/object/cli_obj.c

```diff
@@ -2411,8 +2411,6 @@ shard_task_abort(tse_task_t *task, void *arg)
 	int	rc = *((int *)arg);
 
 	tse_task_complete(task, rc);
-	tse_task_list_del(task);
-	tse_task_decref(task);
```

Comment:
hmm, seems cannot remove above 2 lines?
see the usage in cli_obj.c, obj_req_fanout(),  "tse_task_list_traverse(task_list, shard_task_abort, &rc);"
when insert the task to the list -
2815 >       >       tse_task_addref(shard_task);                                    
2816 >       >       tse_task_list_add(shard_task, task_list);

So the above two lines are to revert L2815 ~ 2816

Reply:
just FYI, the tse_task_complete () will operate one dtp_list, but tse_task_list_del() will operate on dtp_task_list which is defined specially for user's own task list insert/delete.

Reply:
Hmm. Good point @liuxuezhao!

Reply:
well from my testing the tse_task_complete now should be doing those 2 things.
maybe i missed something on the task lists that's different here. i will check more.

Reply:
fwiw if i don't remove them, we get fault injection errors. so that's why i had to. but i will check if there is a better way

Reply:
dtp_task_list is only used for user's own task list.
"we get fault injection errors" not sure which kind of FI error, maybe some other issue in testing code?

Reply:
I think it should be ok now with latest push.